src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Why/DependencyGraphFinder.cs

```diff
@@ -31,28 +31,36 @@ internal static class DependencyGraphFinder
             var dependencyGraphPerFramework = new Dictionary<string, List<DependencyNode>?>(assetsFile.Targets.Count);
             bool doesProjectHaveDependencyOnPackage = false;
 
+            // add null to the list of runtime identifiers to account for projects that do not have a runtime identifier
+            var runtimeIdentifiers = assetsFile.PackageSpec.RuntimeGraph.Runtimes.Keys
+                .Append(null)
+                .ToList();
             // get all top-level package and project references for the project, categorized by target framework alias
             Dictionary<string, List<string>> topLevelReferencesByFramework = GetTopLevelPackageAndProjectReferences(assetsFile, userInputFrameworks);
 
             if (topLevelReferencesByFramework.Count > 0)
             {
                 foreach (var (targetFrameworkAlias, topLevelReferences) in topLevelReferencesByFramework)
                 {
-                    LockFileTarget? target = assetsFile.GetTarget(targetFrameworkAlias, runtimeIdentifier: null);
-
-                    // get all package libraries for the framework
-                    IList<LockFileTargetLibrary>? packageLibraries = target?.Libraries;
-
-                    // if the project has a dependency on the target package, get the dependency graph
-                    if (packageLibraries?.Any(l => l?.Name?.Equals(targetPackage, StringComparison.OrdinalIgnoreCase) == true) == true)
-                    {
-                        doesProjectHaveDependencyOnPackage = true;
-                        dependencyGraphPerFramework.Add(targetFrameworkAlias,
-                                                        GetDependencyGraphForTargetPerFramework(topLevelReferences, packageLibraries, targetPackage));
-                    }
-                    else
+                    foreach (var runtimeIdentifier in runtimeIdentifiers)
                     {
-                        dependencyGraphPerFramework.Add(targetFrameworkAlias, null);
+                        LockFileTarget? target = assetsFile.GetTarget(targetFrameworkAlias, runtimeIdentifier: runtimeIdentifier);
```

Comment:
This is preexisting code so you don't have to change it, but just some general thoughts that I'll create an issue for. 
I don't think this code is doing a very good job of using nullability. 
It's basically treating everything as nullable, but there are situations where the results not be null. 

For example, we should check whether target is null, and once it's not null, things like Libraries shouldn't be null.

Reply:
Yes, I did not change much of the code and there is a lot of objects that are being treated as nulls